shared/selectors.go

```diff
@@ -2,15 +2,9 @@ package shared
 
 import (
 	"github.com/ipld/go-ipld-prime"
-	basicnode "github.com/ipld/go-ipld-prime/node/basic"
-	"github.com/ipld/go-ipld-prime/traversal/selector"
-	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
+	selectorparse "github.com/ipld/go-ipld-prime/traversal/selector/parse"
 )
 
-// entire DAG selector
-func AllSelector() ipld.Node {
-	ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype.Any)
-	return ssb.ExploreRecursive(selector.RecursionLimitNone(),
-		ssb.ExploreAll(ssb.ExploreRecursiveEdge())).
-		Node()
-}
+// AllSelector is a compatibility alias for an entire DAG non-matching-selector
+// Use selectorparse.CommonSelector_ExploreAllRecursively directly in new code
+func AllSelector() ipld.Node { return selectorparse.CommonSelector_ExploreAllRecursively }
```

Comment:
can't we just delete this now or are we trying to do a gentle deprecation?

Reply:
Yeah I'd say keep it and mark it deprecated: https://github.com/golang/go/wiki/Deprecated

There's nearly zero cost to doing that, given how tiny the code is.

Reply:
@mvdan could you do a "code-suggestion" right here? I am not well versed in the right etiquette for deprec. comments...

Reply:
Also, FYI, you can also deprecate the entire package.

Reply:
> you can also deprecate the entire package.

mmmm it has other functions ( different files ), this doesn't apply here does it?